ui/pages/confirmation/confirmation.js

```diff
@@ -258,14 +262,24 @@ export default function ConfirmationPage({
     return INPUT_STATE_CONFIRMATIONS.includes(type);
   };
 
-  const handleSubmit = () =>
-    templateState[pendingConfirmation.id]?.useWarningModal
+  const handleSubmit = async () => {
+    setLoading(true);
+    const submitResult = await (templateState[pendingConfirmation.id]
+      ?.useWarningModal
       ? setShowWarningModal(true)
       : templatedValues.onSubmit(
           hasInputState(pendingConfirmation.type)
             ? inputStates[MESSAGE_TYPE.SNAP_DIALOG_PROMPT]
             : null,
-        );
+        ));
+
+    if (submitResult?.error) {
```

Comment:
We're missing this "submit result error handling" step for confirmations that include a warning modal. For the warning modal case, `onSubmit` is called in the `onSubmit` prop passed to `ConfirmationWarningModal` below.

Perhaps we can move this result handling to a separate function/wrapper function, and use it in both places?